lib/distro.py

```diff
@@ -57,10 +57,10 @@ def build_packages(self, packages):
         """
         self.package_builder.initialize()
         for package in packages:
+            # The package unlock is done inside package_builder.build()
             package.lock()
             package.download_files(recurse=False)
             self.package_builder.build(package)
-            package.unlock()
```

Comment:
I agree with that, but I don't want to break the build function in too many steps and make the Mock class' usability harder. I've decided to externalize the `Mock._prepare()` function, with the small drawback that the build_files copying will be included in the lock.